tests/Library/Vanilla/DateFilterSchemaTest.php

```diff
@@ -288,4 +298,95 @@ public function provideDateFilterFields() {
             ],
         ];
     }
+
+    /**
+     * Provider data for testDateFilterRange.
+     */
+    public function provideDateFilterRange() {
+        $schema = new DateFilterSchema();
```

Comment:
If you're going to use an object like this, it's probably better to move it into your test, instead of your data provider. You can then pass whatever parameters to your test function and hand them down to the `validate` method while keeping everything local to `testDateFilterRange` (avoiding the potential for object pollution).

Reply:
I've found `new DateFilterSchema();` almost in every method in this file. I don't see how my new method is different from others.

Reply:
It isn't about not instantiating a new `DateFilterSchema` instance. I'm saying you definitely should, but it should be moved to your test, instead of your data provider. Right now, you have one object instance being shared across all your tests, because you're putting it in your data provider. They do two very different things. If you put it in your test, you wouldn't have the potential for object pollution that you do now, since the object will be new with each test.

For example, an array row might be changed to:

```php
'Equal (Date)' => [
                '2017-01-01',
                [
                    'startDate' => new DateTimeImmutable('2017-01-01 00:00:00'),
                    'endDate' => new DateTimeImmutable('2017-01-01 23:59:59'),
                    'exclude' => false
                ]
            ]
```

And the test method would become:

```php
    public function testDateFilterRange(string $dateString, array $expectedResult) {
        $schema = new DateFilterSchema();
        $data = $schema->validate($dateString);
        $result = DateFilterSchema::dateFilterRange($data);
        $this->assertEquals($expectedResult, $result);
    }
```